### PR TITLE
New Okta Virustotal Greynoise steps

### PR DESCRIPTION
**Pull request by hagaishapira**

### CODE REVIEW

File: steps/okta/user/unsuspend/main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		return step.ExitCodeFailure, nil, err
+	}
+
+	//send request
+	_, resp, err := oktaClient.User.ActivateUser(s.UserId, nil)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return step.ExitCodeOK, []byte(stepResultInvalidID), nil
+		} else if resp.StatusCode == http.StatusForbidden {
+			return step.ExitCodeOK, []byte(stepResultInvalidState), nil
```
Comment (eldadru) [inline steps/okta/user/unsuspend/main.go:54]: Should't that be an error with proper message explaining we don't have permissions?

File: steps/okta/user/unsuspend/main.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	oktaClient, err := okta.NewClient(s.ctx, okta.WithOrgUrl(fmt.Sprintf("https://%s", s.OktaDomain)), okta.WithToken(s.OktaApiToken))
+	if err != nil {
+		return step.ExitCodeFailure, nil, err
+	}
+
+	//send request
+	_, resp, err := oktaClient.User.ActivateUser(s.UserId, nil)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return step.ExitCodeOK, []byte(stepResultInvalidID), nil
```
Comment (eldadru) [inline steps/okta/user/unsuspend/main.go:52]: Same, should be an error with proper error message

File: steps/okta/user/suspend/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  description: "Suspends a user."
+  vendor: okta
+  type: remediation
+
+envs:
+  - name: OKTA_API_TOKEN
+    description: "An Okta API token that is used to authenticate requests to Okta APIs."
+    type: "string"
+    required: true
+  - name: OKTA_DOMAIN
```
[the remark this answers is not in the document]
Comment (hagaishapira) [reply steps/okta/user/suspend/manifest.yaml:15]: image

[the remark this answers is not in the document]
Comment (eldadru) [reply steps/okta/user/suspend/manifest.yaml:15]: Gotcha

File: steps/okta/user/unsuspend/main.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		return step.ExitCodeFailure, nil, err
+	}
+
+	//send request
+	_, resp, err := oktaClient.User.ActivateUser(s.UserId, nil)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return step.ExitCodeFailure, []byte(stepResultInvalidID), fmt.Errorf("invalid user id: user id doesn't exist")
+		} else if resp.StatusCode == http.StatusForbidden {
+			return step.ExitCodeFailure, []byte(stepResultInvalidState), fmt.Errorf("invalid user state: user is not in the SUSPENDED state")
```
[the remark this answers is not in the document]
Comment (hagaishapira) [reply steps/okta/user/unsuspend/main.go:54]: image

**Review by eldadru — APPROVED** [review]

**Pull request merged**